pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -319,22 +315,6 @@ def on_validation_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModul
             return
         self.save_checkpoint(trainer)
 
-    def on_train_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
```

Comment:
This is an ancient artifact of the days we had a manual `_check_checkpoint_callback` call in the `Trainer`.

It's not necessary anymore because we will have already saved for this global step in either `on_validation_end` or in `on_train_epoch_end`, so `should_skip_saving_checkpoint = True` and this never runs.